versioning.py

```diff
@@ -100,5 +100,22 @@ def do_versioning(self):
                     )
                 )
                 c.is_reference = True
+        if version < (0, 27, 4):
```

Comment:
A few things to note:
- Imports can be avoided, constraints can be checked by constraint.type == "DISTANCE", entities by using the "is_xxx" methods (e.g. entity.is_line() and entity.is_2d()).
- In general it's nice to avoid deep nesting by inverting the if checks in a loop, just do:
    if c.type != "DISTANCE": continue

Reply:
I am not sure about only checking constraint.type == "DISTANCE", entities by using the "is_xxx" methods (e.g. entity.is_line() and entity.is_2d()). 
This bit of code can change quite a lot in a users drawing. I went from the assumption that you cannot know what weird stuff happened in their drawing and everything needs to be checked first. I think this is the only way to be sure? I also added an extra length check on line_dependencies just to cover such cases. 

Correct about the nested if statements, I pushed an update.

Reply:
I'm not saying you should remove checks, just  to replace them with the corresponding method calls. This can help with readability and make the imports unnecessary. 

Anyway this was just a note, i'm fine with either way. :)